client/src/components/FavoriteButton.tsx

```
@@ -0,0 +1,69 @@
+    <button
+      onClick={handleAction}
+      disabled={isLoading}
+      className={`absolute top-4 right-4 p-2 rounded-full transition-colors duration-200 
+        ${
+          isDeleteMode
+            ? 'bg-red-500 hover:bg-red-600'
+            : 'bg-white/20 backdrop-blur-sm hover:bg-white/30'
+        }`}
+    >
+      {isLoading ? (
+        <span className="loading loading-spinner loading-sm"></span>
```

Biome (1.9.4)

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)